Revive image uploader

okkez opened the pull request.

okkez committed:
Recreate image model

okkez committed:
Revive carrierwave

This commit is based on 73cdf861d1ed888c15a86a7cd579e87dc65ec709.

okkez committed:
Revive image uploader on ActiveAdmin

tricknotes commented:
Is `Image` -> `Site` relation unnecessary?

okkez commented:
I'm not sure about that.

okkez committed:
Set relation between Site and Image

Destroy images related to a site when it is destroyed.

okkez committed:
Revert "Set relation between Site and Image"

This reverts commit 4a299dcaa44a15aceaba55409a4acd8a000bc389.

okkez committed:
Upload multiple images related to post

TODO
  * Display smaller image on form
  * Display markdown snippet on form

okkez committed:
Break a long line

okkez committed:
Stop to display empty image

okkez committed:
Display image URL after image

okkez committed:
Display small image as a hint

okkez committed:
Add test to destroy site

tricknotes commented on test/models/site_test.rb:
```diff
@@ -1,7 +1,23 @@
+                        css_url: "http://example.com/application.css")
+    @post = @site.posts.create(title: "title", body: "body", original_id: 1)
+    Tempfile.open do |file|
+      @post.images.create(image: file)
```
`create!` だと保存に失敗した時に例外を投げてくれるので安心感あります！

okkez replied:
なるほど！！！

okkez committed:
Use `create!` instead of `create`

Because `create!` raises an error when failed to create record.

tricknotes commented on app/uploaders/image_uploader.rb:
```diff
@@ -0,0 +1,11 @@
+class ImageUploader < CarrierWave::Uploader::Base
+  if fog_credentials.present?
```
This configuration seems a daimon specific config, (not only each uploader)
If so, should it be moved to `ApplicationUploader`, `config/initializers/carrierwave.rb` or somewhere?

okkez replied:
We don't need to move that configuration because we have only one uploader, for now.

tricknotes replied:
I see :+1:

okkez committed:
Remove null constraint from posts.original_id

Because we create new post from ActiveAdmin and the post doesn't have
original_id.

okkez committed:
Remove original_id

original_id is optional.

tricknotes commented:
なるほどー！記事に対して画像を設定していく導線いい感じですね！
とすると、 `/admin/images` では `index` だけでよかったりしますか？(もしかすると `/admin/images` 自体なくてもよかったりしますか？)

okkez commented:
ダッシュボードに images なくてもいいかもですねー。

okkez committed:
Add null constraint to images.post_id

okkez committed:
Remove image from dashboard

We can manage images from posts.

okkez commented:
Close #149 
Close #166

tricknotes commented:
I thinks this is wonderful :dancers:
